### DISCUSSION TITLE
Testing for remix application

### DISCUSSION THREAD

**Discussion by Karma-Tsering** [remix-run/remix#9723] [Q&A]
I have been looking for a unit testing in remix and I found jest and vitest but I was not able to setup and work  on it. I didn't found any document regarding the same. Kindly guide me to do the unit testing for remix application.

**Comment by jrestall-finact** [comment]
You can see an example with vitest at [epicweb-dev/epic-stack](https://github.com/epicweb-dev/epic-stack/blob/main/app/routes/users%2B/%24username.test.tsx) or [epicweb-dev/full-stack-testing/tree/main/exercises/07.remix-component/01.problem.stub-remix](https://github.com/epicweb-dev/full-stack-testing/tree/main/exercises/07.remix-component/01.problem.stub-remix)

Remix is stubbed in unit tests using `createRemixStub`:
https://remix.run/docs/en/main/utils/create-remix-stub
